src/main/java/de/dennisguse/opentracks/fragments/FilterDialogFragment.java

```diff
@@ -0,0 +1,186 @@
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AlertDialog;
+import androidx.fragment.app.DialogFragment;
+import androidx.fragment.app.FragmentManager;
+
+import com.google.android.material.textfield.TextInputEditText;
+
+import java.util.ArrayList;
+import java.util.Calendar;
+
+import de.dennisguse.opentracks.R;
+import de.dennisguse.opentracks.content.data.Track;
+import de.dennisguse.opentracks.util.CalendarUtil;
+
+public class FilterDialogFragment extends DialogFragment {
+
+    private static final String TAG = FilterDialogFragment.class.getSimpleName();
+    public static final String KEY_FILTER_ITEMS = "filterItems";
+
+    private FilterDialogListener filterDialogListener;
+    private ArrayList<FilterItem> filterItems = new ArrayList<>();
+
+    public static void showDialog(FragmentManager fragmentManager) {
+        FilterDialogFragment filterDialogFragment = new FilterDialogFragment();
+        filterDialogFragment.show(fragmentManager, TAG);
+    }
+
+    public static void showDialog(FragmentManager fragmentManager, ArrayList<FilterItem> items) {
+        Bundle bundle = new Bundle();
+        bundle.putParcelableArrayList(KEY_FILTER_ITEMS, items);
+
+        FilterDialogFragment filterDialogFragment = new FilterDialogFragment();
+        filterDialogFragment.setArguments(bundle);
+        filterDialogFragment.show(fragmentManager, TAG);
+    }
+
+    @NonNull
+    @Override
+    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
+        filterItems = getArguments().getParcelableArrayList(KEY_FILTER_ITEMS);
+
+        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
+        builder.setTitle(getString(R.string.generic_filter));
+
+        LayoutInflater inflater = LayoutInflater.from(getActivity());
+        View layout = inflater.inflate(R.layout.fragment_filter_dialog, null, false);
+        GridLayout itemsLayout = layout.findViewById(R.id.filter_items);
+        builder.setView(layout);
+
+        for (FilterItem item : filterItems) {
+            View view = inflater.inflate(R.layout.fragment_filter_dialog_item, null);
+
+            CheckBox checkBox = view.findViewById(R.id.filter_dialog_check_button);
+            checkBox.setText(item.value);
+            checkBox.setChecked(item.isChecked);
+            checkBox.setTag(item.id);
+            checkBox.setOnClickListener(v -> item.isChecked = !item.isChecked);
+
+            itemsLayout.addView(view);
+        }
+
+        DatePicker datePickerFrom = layout.findViewById(R.id.filter_date_picker_from);
+        DatePicker datePickerTo = layout.findViewById(R.id.filter_date_picker_to);
+        TextInputEditText dateFrom = layout.findViewById(R.id.filter_date_edit_text_from);
+        TextInputEditText dateTo = layout.findViewById(R.id.filter_date_edit_text_to);
+
+        CalendarUtil calendarFirstWeekDay = CalendarUtil.ofFirstDayOfTheWeek();
```

Comment:
somehow this feels complicated.
No better suggestion yet :)

Reply:
I'm going to think about this.